chore(docs): README refactor (#1)
bee344/asset-hub-examples: README.md

The Asset Hub will use DOT as its native currency. Users can transfer DOT from the Relay Chain into the Asset Hub and use it natively, and to the Relay Chain Asset Hub back to the Relay Chain for staking, governance, or any other activity.

### Running a Polkadot Asset Hub node

Using the Asset Hub will require running a parachain node to sync the chain. This is very similar to running a Polkadot node, with the addition of some extra flags. You can download the latest release of the [`polkadot-parachain` binary](https://github.com/paritytech/polkadot-sdk/releases/latest/) or build it from [source](https://github.com/paritytech/polkadot-sdk/) with the following commands:
```bash
$ cargo build --release --locked --bin polkadot-parachain
```

Another alternative is using Docker, this is more advanced, so it's best left up to those already familiar with docker or who have completed the other set-up instructions:
```bash
$ docker run -p 9944:9944 -p 9615:9615 parity/polkadot:latest \
\ --chain asset-hub-westend
\ --rpc-external
\ --prometheus-external
```
And then run the node with:
```bash
$ ./target/release/polkadot-parachain --chain asset-hub-westend
```

> #### Types of Nodes

Reviewer: dont we have any update about the potential different node functioning? like run a archive node for a  parachain with only a full node for relay and so on.

- A **full node** prunes historical states: all finalized blocks' states older than a configurable number except the genesis block's state. This is 256 blocks from the last finalized one by default. A pruned node this way requires much less space than an archive node.

An **archive node** can become a **full node**, but for a **full node** to become an **archive node**, you must first purge your database and resync your node, starting in archive mode.

#### Hardware requirements

Currently there are no hardware requirements specific for the collator nodes, since they don't perform time-critical tasks. The only requirement is to have enough storage for the type of node intended, which can be retrieved from [here](https://stakeworld.io/docs/dbsize). Other than that, any relatively performant equipement or any cloud provider will suffice. You can also look into the [reference hardware](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#reference-hardware) for validators, but be aware that these will probably be overkill for a non-validator node.

#### NOS

> NoS stands for Node + (API) Sidecar and consists of a simple script to launch a full node of [Asset Hub](https://wiki.polkadot.network/docs/learn-system-chains#asset-hub) together with an instance of [Sidecar](https://github.com/paritytech/substrate-api-sidecar) on a machine (using Docker).

Reviewer: remove link to Asset Hub here.

To start using NoS you can follow this [guide](https://github.com/paritytech/nos?tab=readme-ov-file#using-nos). For generating an instance of Polkadot Asset Hub you can follow this [example `.env` file](https://github.com/paritytech/nos/blob/main/.env.statemint.example), just updating `POLKADOT_VERSION` and `POLKADOT_PARACHAIN_VERSION`.

NoS downloads and maintains archive nodes for at least one Relay Chain and one parachain, in this case Polkadot Relay Chain and the Polkadot Asset Hub respectively, so this will affect the amount of storage needed. At time of writing, the database sizes are 1.7 TiB for the Polkadot Relay Chain node and 133 GiB for the Polkadot Asset Hub node, both running with `pruning=archive`. Sidecar is stateless, so the amount of storage it requires is marginal. You can check the current db size for each type of node [here](https://stakeworld.io/docs/dbsize).

> #### Maintenance

Reviewer: we need to put more datails about monitoring runtime upgrades. As said in my doc, something like [this](https://wiki.polkadot.network/docs/learn-runtime-upgrades#monitoring-runtime-changes) (or just linking it adequately) for Asset Hub

It's good practice to keep your node up to date with the latest version, which you can download from [here](https://github.com/paritytech/polkadot-sdk/releases). It's also recommended to keep the tools you are using up to date. Lastly, we recommend keeping track of the Polkadot Fellowship's [runtime upgrades](https://github.com/polkadot-fellows/runtimes/releases/latest) to be aware of critical logic changes. These runtime upgrades are voted on and executed via OpenGov, and the proposals with their enactment dates and other details can be found [here](https://polkadot.polkassembly.io/whitelisted-caller?trackStatus=all&page=1).

## Foreign Assets in Polkadot Asset Hub

> Foreign assets are those assets in Asset Hub whose native blockchain is not Asset Hub. These are mainly native tokens from other parachains or bridged tokens from other consensus systems (such as Ethereum). Once a foreign asset has been registered in Asset Hub (by its root origin), users are enabled to send this token from its native blockchain to Asset Hub and operate with it as if it was any other asset.

Reviewer: Add a paragraph above like:

"Assets, both native and foreign, are stored as a map from an ID to information about the asset, including a management team, total supply, total number of accounts, its sufficiency for account existence, and more. Additionally, the asset owner can register metadata like the name, symbol, and number of decimals for representation.

Some assets, as determined by on-chain governance, are regarded as “sufficient”. Sufficiency means that the asset balance is enough to create the account on-chain, with no need for the DOT existential deposit. Likewise, you cannot send a non-sufficient asset to an account that does not exist. Sufficient assets can be used to pay transaction fees (i.e. there is no need to hold DOT on the account). Assets do have a minimum balance, and if an account drops below that balance, the dust is lost. 

For an more details on general asset management in Asset Hub, refer to [the assets documentation](https://wiki.polkadot.network/docs/build-integrate-assets#assets-basics) of the Polkadot Wiki."

Author: Cool, at first that paragraph was in the guide but I was in doubt on whether we wanted the guide to be more or less constricted to just foreignAssets, so I removed it

> Practically speaking, foreign assets are handled by the `foreign-assets` pallet in Asset Hub, which is an instance of the previously mentioned `assets` pallet. Hence, this pallet exposes the same interface to users and other pallets as `assets` pallet. For example, to transfer a certain amount of a foreign asset (id) to an account (target), this pallet exposes the following call: 

Reviewer: ```suggestion
Practically speaking, foreign assets are handled by the `foreign-assets` pallet in Asset Hub, which is an instance of [the `assets` pallet](https://marketplace.substrate.io/pallets/pallet-assets/). Hence, this pallet exposes the same interface to users and other pallets as `assets` pallet. For example, to transfer a certain amount of a foreign asset (id) to an account (target), this pallet exposes the following call: 
```

``` ts
foreignAssets.transferKeepAlive(id, target, amount)
```

The main difference to take into account for foreign assets is their identifier. Unlike using integers as identifiers in the `assets` pallet, assets stored in the `foreign-assets` pallet are identified by [their XCM multilocation](https://wiki.polkadot.network/docs/learn/xcm/fundamentals/multilocation-summary). Taking the example below, the id input parameter of the call above will be a multilocation type as `{parents: 1, interior: {X1: {Parachain: 2,123}}}`.

### Foreign Asset Operations

> The main functions you will probably interact with are `foreignAssets.transfer_allow_death` and `foreignAssets.transfer_keep_alive`. These functions transfer some `amount` (balance) of an `AssetId` (a `u32`, not a contract address) to another account.

Reviewer: ```suggestion
The main functions you will probably interact with are `foreignAssets.transfer_allow_death` and `foreignAssets.transfer_keep_alive`.

The `foreign-assets` also provides an `approve_transfer`, `cancel_approval`, and `transfer_approved` interface for non-custodial operations.

> Note that you can use the same addresses (except [pure proxies](https://wiki.polkadot.network/docs/learn-proxies-pure#anonymous-proxy-pure-proxy)!) on the Asset Hub that

Reviewer: ```suggestion
Note that you can use the same addresses (except [pure proxies](https://wiki.polkadot.network/docs/learn-proxies-pure#anonymous-proxy-pure-proxy)!) on Asset Hub that you use on the Relay Chain. The SS58 encodings are the same; only the chain information (genesis hash, etc.) will change on transaction construction.
```

> you use on the Relay Chain. The SS58 encodings are the same; only the chain information (genesis hash, etc.) will change on transaction construction.

Reviewer: ```suggestion
```

A great resource to learn more about Error Management in XCM is the Polkadot blog post from Dr. Gavin Wood, [XCM Part III: Execution and Error Management](https://www.polkadot.network/blog/xcm-part-three-execution-and-error-management).

> ### Running a Polkadot Asset Hub node

Reviewer: Line 90 of the new version is wrong, we are missing teleport to be mentioned:

Suggestion:

"Currently, foreign assets can be sent and received in the Asset Hub either with a [Reserve Backed Transfer or Teleporting](https://wiki.polkadot.network/docs/learn-xcm-pallet#transfer-reserve-vs-teleport) from other parachains. In both cases, the event emitted when processing the transfer is the `foreignAssets.issued` event. Hence, you should listen to these events, pointing to an address in their system. For this, you must query every new block created, loop through the events array, filter for any `foreignAssets.issued` event, and apply the appropriate business logic."

> 

Reviewer: Line 97 is wrong, the relay chain is not queried.

Reviewer: please double check that the info of lines 97 and 98 also applies for asset hub when receiving XMC.

> Using the Asset Hub will require running a parachain node to sync the chain. This is very similar to running a Polkadot node, with the addition of some extra flags. You can download the latest release of the [`polkadot-parachain` binary](https://github.com/paritytech/polkadot-sdk/releases/latest/) or build it from [source](https://github.com/paritytech/polkadot-sdk/) with the following commands:

Reviewer: Line 105 of new version and other ocurrences: Do not mention system parachains, always talk about Asset Hub

}
```

> Using the generic endpoints for the pallets, sidecar can also retrieve values specific to the configuration of the pallet, such as `constants`, `dispatchables`, `errors`, `events` and `storage`.

Reviewer: Line 139 above and other occurrences: let's call it Sidecar or API Sidecar. Not "the sidecar" etc.

Reviewer: I would remove Line 179 and everything below and add it to the PJS example, Here it is enough if we mention how to submit a transaction (already signed and ready, a byte array). You can reference the relevant example here.

It exposes a reduced set of methods which facilitates users to send transfers to other (para) chains or locally.
It covers the pallet `assets` and its instances `pool-assets` and `foreign-assets`, as well as local transactions.

> #### Tx Wrapper Polkadot

Reviewer: Remove this section and any txwrapper mention if there is no foreign assets support on time.
